.github/workflows/Tests.yaml

```diff
@@ -329,6 +330,9 @@ jobs:
         working-directory: /work/build
         run: |
           make -j2 RunTests
+          # Find and delete object files in the build dir for unit tests.
+          # This reduces the size of the build dir by at least 1.5GB
+          find ./tests/ -type f -name "*.cpp.o" -print0 | xargs rm
```

Comment:
[optional] `find` has an `-exec` flag which is probably better than piping in generally, but I think that only has an impact on pathological filenames with special characters and spaces

Reply:
Currently we use `find` with the pipe in a number of places in the code, so I think if we were to change this we should do so universally (outside of this PR since I'd like CI to be reliably again :P ). Does that sound okay?

Reply:
Yep, sounds fine. :+1:

Reply:
With `-print0` weird file names should be safe.  There's a race condition with xargs that can be a security problem if your code needs to worry about that, but there's always a race condition with xargs and you generally don't care.

However, doesn't the `-print0` have to be matched by a `-0` to xargs?

Reply:
@wthrowe yep you're right! Good catch :) Hopefully this will save us another ~2GB on debug builds then.